src/main/java/blob/Parser.java

```diff
@@ -0,0 +1,258 @@
+                    s.append("Noted, I've removed this task:\n");
+                    s.append(t + "\n");
+                    s.append("Now you have " + tasklist.size() + " tasks in the list.");
+                } catch (NumberFormatException e) {
+                    s.append("Invalid Command!");
+                } catch (IndexOutOfBoundsException e) {
+                    int taskNum = Integer.parseInt(arr[i + 1]);
+                    s.append(String.format("There is no task at task number '%d'", taskNum));
+                }
+                return s.toString();
+
+            } else if (act.equals("todo")) {
+                // task name error handling
+                if (arr.length == 1) {
+                    return "Sorry, I am unable to generate an empty 'todo' task!";
+                }
+
+                StringBuilder msg = new StringBuilder("");
+
+                // for task name string
+                StringBuilder a = new StringBuilder(arr[i + 1]);
+                for (int j = i + 2; j < arr.length; j++) {
+                    StringBuilder str = new StringBuilder(" " + arr[j]);
+                    a = a.append(str);
+                }
+
+                Todo t = new Todo(a.toString(), false);
+                tasklist.addTask(t);
+                msg.append("Got it. I've added this task:\n");
+                msg.append(t + "\n");
+                msg.append("Now you have " + tasklist.size() + " tasks in the list.");
+                return msg.toString();
+
+            } else if (act.equals("deadline")) {
+                int by = 0;
+                for (int j = 0; j < arr.length; j++) {
+                    if (arr[j].equals("/by")) {
+                        by = j;
+                    }
+                }
+
+                // task name error handling
+                if (by <= 1) {
+                    return "I require a description for your 'deadline' task 😅!";
+                }
+
+                // task deadline error handling
+                if (by == arr.length - 1) {
+                    return "Your deadline task can't not have a deadline! Please enter your task again!";
+                }
+
+                //task deadline only having date or time
+                if (by == arr.length - 2) {
+                    return "You need to add both DATE and TIME to your deadline \uD83D\uDD27! (Format: yyyy-mm-dd HH:mm)";
+                }
+
+                StringBuilder msg = new StringBuilder("");
+
+                // for task name string
+                StringBuilder a = new StringBuilder(arr[i + 1]);
+                for (int l = i + 2; l < by; l++) {
+                    StringBuilder str = new StringBuilder(" " + arr[l]);
+                    a = a.append(str);
+                }
+
+                // for task deadline string
+                StringBuilder sDate = new StringBuilder(arr[by + 1]);
+                StringBuilder sTime = new StringBuilder(arr[by + 2]);
```

Comment:
Maybe can make the name become more explainable to what it do, like sDate can become strDate or stringDate